core/services/gateway/handlers/functions/subscriptions.go

```diff
@@ -149,15 +149,13 @@ func (s *onchainSubscriptions) queryLoop() {
 		}
 
 		if start > lastKnownCount {
+			updateLastKnownCount()
```

Comment:
I find this code hard to read since `updateLastKnownCount()` is a function with the side effect.

Maybe instead you could refactor to use a function called `getLastKnownCount` instead?
`lastKnownCount := getLastKnownCount()`

This makes more sense to me, but maybe your implementation is just a common pattern in Go that I am not familiar with? I'd appreciate thoughts from others here.

Reply:
Done